src/cli/input-from-terminal.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+const readline = require("readline");
+const { stdin } = require("node:process");
+
+/**
+ * This function starts reading any input which user types on the console.
+ * After the user clicks ctrl+D to exit, the input is formatted and printed
+ * on the screen.
+ * @returns {Promise<string>} On success, resolves the input code joined together
+ * to form a single string. On error, reject with an error message
+ */
+function getInputFromTerminal() {
+  return new Promise((resolve) => {
+    const inputCode = [];
+    const readLine = readline.createInterface({ input: stdin });
+    readLine.on("line", (line) => {
+      inputCode.push(line);
+    });
+    readLine.on("close", () => {
```

Comment:
Can't exit with <kbd>Ctrl</kbd> + <kbd>D</kbd>, I also tried [<kbd>Ctrl</kbd> + <kbd>Z</kbd>](https://stackoverflow.com/questions/16136400/how-to-send-eof-via-windows-terminal), can't either.

Reply:
Fixed in [8d2872f6de17859b5d474cff88c1b1806b3895a6](https://github.com/prettier/prettier/pull/13428/commits/8d2872f6de17859b5d474cff88c1b1806b3895a6)